### ISSUE TITLE
SA.js RAM dump broken

### ISSUE THREAD

**Issue by RossComputerGuy** [Ross-Technologies/SherwoodArch#4]
Log:
```

<--- Last few GCs --->

[3575:0x3ed7c20]     4252 ms: Scavenge 389.0 (394.0) -> 388.0 (394.0) MB, 0.1 / 0.0 ms  allocation failure 
[3575:0x3ed7c20]     4276 ms: Scavenge 389.0 (394.0) -> 388.0 (394.0) MB, 0.2 / 0.0 ms  allocation failure 
[3575:0x3ed7c20]     4300 ms: Scavenge 389.0 (394.0) -> 388.0 (394.0) MB, 0.2 / 0.0 ms  allocation failure 
[3575:0x3ed7c20]     4325 ms: Scavenge 389.0 (394.0) -> 388.0 (394.0) MB, 0.1 / 0.0 ms  allocation failure 


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x37f6919a9891 <JS Object>
    1: DoJoin(aka DoJoin) [native array.js:~96] [pc=0x1ec9249997d8](this=0xdb6f4f82311 <undefined>,p=0x1ccb086c2641 <a Float64Array with map 0x24bc6c387271>,q=1073741824,E=0xdb6f4f82421 <false>,A=0x37f6919c4889 <String[1]: ,>,z=0xdb6f4f82421 <false>)
    3: Join(aka Join) [native array.js:1] [bytecode=0x236638082341 offset=86](this=0xdb6f4f82311 <undefined>,p=0x1ccb086c2641 <a Float64Array with...

FATAL ERROR: invalid table size Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x1363f0c [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Dictionary<v8::internal::SeededNumberDictionary, v8::internal::SeededNumberDictionaryShape, unsigned int>::EnsureCapacity(v8::internal::Handle<v8::internal::SeededNumberDictionary>, int, unsigned int) [node]
 6: v8::internal::Dictionary<v8::internal::SeededNumberDictionary, v8::internal::SeededNumberDictionaryShape, unsigned int>::Add(v8::internal::Handle<v8::internal::SeededNumberDictionary>, unsigned int, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyDetails, int*) [node]
 7: v8::internal::SeededNumberDictionary::AddNumberEntry(v8::internal::Handle<v8::internal::SeededNumberDictionary>, unsigned int, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyDetails, v8::internal::Handle<v8::internal::JSObject>) [node]
 8: 0xe7b896 [node]
 9: v8::internal::JSObject::AddDataElement(v8::internal::Handle<v8::internal::JSObject>, unsigned int, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow) [node]
10: v8::internal::Object::AddDataProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow, v8::internal::Object::StoreFromKeyed) [node]
11: v8::internal::Object::SetProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::LanguageMode, v8::internal::Object::StoreFromKeyed) [node]
12: v8::internal::Runtime::SetObjectProperty(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::LanguageMode) [node]
13: v8::internal::Runtime_KeyedStoreIC_Slow(int, v8::internal::Object**, v8::internal::Isolate*) [node]
14: 0x1ec92480437d
Aborted (core dumped)
```

**Comment by RossComputerGuy** [comment]
The RAM dump may have to be written in segments.

**Comment by RossComputerGuy** [comment]
Fixed in a945e71ae177f79474906984fbaa3bb41b068648
